lib/src/dropdown/dropdown.component.ts

```diff
@@ -0,0 +1,503 @@
+import {
+    Component,
+    Input,
+    forwardRef,
+    OnChanges,
+    ViewChild,
+    ElementRef,
+    ViewChildren,
+    QueryList,
+    OnDestroy,
+    NgZone,
+    SimpleChanges,
+    Provider,
+} from "@angular/core";
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from "@angular/forms";
+import { fromEvent, Subscription } from "rxjs";
+
+export interface Item {
+    /** The label or text to be displayed in the list */
+    label: string;
+    /** any value which should be tied to the item */
+    value: any;
+    /** The id or the unique key of the item */
+    key: string;
+}
+
+interface UniqueItem {
+    id: string;
+    optionItem: Item;
+    selected: boolean;
+}
+
+interface DisplayItem extends UniqueItem {
+    className: string;
+}
+
+const CUSTOM_DROPDOWN_CONTROL_VALUE_ACCESSOR: Provider = {
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => DropdownComponent),
+    multi: true,
+};
+
+@Component({
+    selector: "sebng-dropdown",
+    templateUrl: "./dropdown.component.html",
+    styleUrls: ["./dropdown.component.scss"],
+    providers: [CUSTOM_DROPDOWN_CONTROL_VALUE_ACCESSOR],
+})
+export class DropdownComponent implements ControlValueAccessor, OnChanges, OnDestroy {
+    @Input() name: string;
+    @Input() list: Array<Item>;
+    @Input() id?: string;
+    @Input() label?: string;
+    @Input() error?: string;
+    @Input() placeHolder?: string;
+    @Input() className?: string;
+    @Input() disabled?: boolean = false;
+    @Input() native?: boolean = false;
+    @Input() multi?: boolean;
+    @Input() clearable?: boolean = false;
+    @Input() searchable?: boolean = false;
+    @Input() searchPlaceholder?: string = "";
+    @Input() nativeOnChange?: (event: Item | Array<Item> | UIEvent) => void;
+    @Input() ellipsisMode: boolean;
+    // Placeholders for the callbacks which are later provided
+    // by the Control Value Accessor
+    private onTouchedCallback: () => void;
+    private onChangeCallback: (_: any) => void;
+
+    private _subscriber: Subscription = null;
+
+    private _open = false;
+    private _searchText = "";
+    private _shouldFocus = true;
+    private _currentFocused = -1;
```

Comment:
```suggestion
    private _open: boolean = false;
    private _searchText: string = "";
    private _shouldFocus: boolean = true;
    private _currentFocused: number = -1;
```

Reply:
Can you refactor `_open` to sound like a boolean value. Something like `_isOpen`, also the getter